packages/dds/map/src/mapKernel.ts

```diff
@@ -136,7 +140,7 @@ export class MapKernel {
     /**
      * Keys that have been modified locally but not yet ack'd from the server.
      */
-    private readonly pendingKeys: Map<string, number> = new Map();
+    private readonly pendingKeys: Map<string, IMapLocalOpMetadata> = new Map();
```

Comment:
still reading, but do we only need to keep the latest here. i was thinking we would need to keep the list of all changes.

for example:

- begin order sequentially
  - set key A
  - begin nested order sequentially
    - set key A
    - rollback
 - rollback

Reply:
adding some tests here would be good too. i assume that's the plan, but can help see how the changes work for reviewers as well